### PR TITLE
Notification system for storage changes

### PR DESCRIPTION
**Pull request by mbarnes**
This and https://github.com/projectatomic/commissaire-service/pull/62 are co-dependent.

Will have to verify tests manually with both PRs installed.

### CODE REVIEW

**Comment by mbarnes** [comment]
Getting close to ready...

`commctl` needs some love for the use case I'm playing with.

**Comment by mbarnes** [comment]
I think this is finally ready for review.  I have notifications working here with what I think is a good API.

Requires both this and https://github.com/projectatomic/commissaire-service/pull/62 for tests to pass.

File: Vagrantfile [hunk 1]
```diff
@@ -168,10 +168,6 @@ Vagrant.configure(2) do |config|
       sudo cp /vagrant/commissaire-service/conf/systemd/commissaire-investigator.service /etc/systemd/system/commissaire-investigator.service
       sudo sed -i 's|^ExecStart=.*|ExecStart=/bin/bash -c ". /home/vagrant/commissaire_env/bin/activate \\&\\& commissaire-investigator-service --bus-uri redis://192.168.152.101:6379"|' /etc/systemd/system/commissaire-investigator.service
 
-      echo "===> Setting up commissaire-watcher service to autostart"
```
Comment (ashcrow) [inline Vagrantfile:171]: Why disable ``watcher``?

File: src/commissaire/storage/client.py [hunk 2]
```diff
@@ -32,6 +95,58 @@ def __init__(self, bus_mixin):
...
+        consumer_list = []
+        exchange = self.bus_mixin.producer.exchange
+        for routing_key, callbacks in self.notify_callbacks.items():
+            queue = kombu.Queue(
+                exchange=exchange, routing_key=routing_key)
+            consumer = Consumer(
+                queues=queue, callbacks=callbacks)
+            consumer_list.append(consumer)
+            self.bus_mixin.logger.info(
+                'Listening for "{}" notifications'.format(routing_key))
```
Comment (ashcrow) [inline src/commissaire/storage/client.py]: loggers are the only place ``%s`` should be used if possible.

File: Vagrantfile (hunk 1 above)
Comment (mbarnes) [reply Vagrantfile:171]: It interferes with the E2E tests by intermittently injecting additional notifications on the bus that the tests aren't expecting.  Disabling it until I can figure out a better solution.

File: src/commissaire/storage/client.py (hunk 2 above)
Comment (mbarnes) [reply src/commissaire/storage/client.py]: Right... old habits.

**Comment by ashcrow** [comment]
bot, retest this please

**Comment by ashcrow** [comment]
Retesting post merging.

**Comment by ashcrow** [comment]
Good news and bad news.

Good news: The tests passed.
Bad news: ```nose.plugins.cover: ERROR: TOTAL Coverage did not reach minimum required: 80%```

**Comment by mbarnes** [comment]
@ashcrow :arrow_up: Rebased on latest `master` and tacked on some fixups.

**Review by ashcrow — APPROVED** [review]

**Comment by ashcrow** [comment]
Merged manually.

**Pull request closed without merging**